### ISSUE TITLE
Render problem using MOV + ProRes

### ISSUE THREAD

**Issue by Adenser** [lay295/TwitchDownloader#55]
Hi, I have a problem rendering a JSON chat using mov + ProRes
"ERROR: The pipe has been ended"
with RLE it works properly, its an issue only with ProRes
settings: 
![image](https://user-images.githubusercontent.com/39248856/90539861-b069ab00-e180-11ea-990d-f4b09a0f0613.png)
version 1.30, also tested with 1.20, same issue

**Comment by lay295** [comment]
One day i'll have better error logging, until that day...

But you can download this build that will output the ffmpeg process into the log. Once it errors copy/paste it here and should be easier to tell whats going wrong.
https://drive.google.com/file/d/1cdVYc38dOIve0X07Qk45TfngfvlXqlw8/view?usp=sharing

**Comment by Adenser** [comment]
Have you changed anything between this version and 1.3?
Expect 35GB file everything went fine and chat have rendered itself
![image](https://user-images.githubusercontent.com/39248856/90556258-e31f9d80-e198-11ea-8e81-f7a8d0e3f74c.png)

**Comment by lay295** [comment]
Nothing was changed except for logging. Also the huge file size with ProRes is normal. Does it still error out on the regular 1.3?

**Comment by Adenser** [comment]
There is a chance it was my own fault, I've rendered chat on 1.3 but I've chosen my second drive and render went fine, probably I was out of memory because I was doing it on my sys drive with 24gb space available, did not expect the video to have over 30GBs.

About the file size, for sure it's not normal, ProRes has profiles that set max bitrate, https://ffmpeg.org/ffmpeg-codecs.html#Private-Options-for-prores_002dks If there is no profile defined by default ProRes chose a 4:4:4:4 profile, I'm not a developer but I would think about setting profile to 0 for rendering (max bitrate for twitch is 6mpbs, so it shouldn't hurt quality at all)
-1: auto (default)
0: proxy ≈ 45Mbps YUV 4:2:2
1: lt ≈ 102Mbps YUV 4:2:2
2: standard ≈ 147Mbps YUV 4:2:2
3: HQ ≈ 220Mbps YUV 4:2:2
4: 4444≈ 330Mbps YUVA 4:4:4:4
5: 4444xq ≈ 500Mbps YUVA 4:4:4:4

**Comment by lay295** [comment]
The main point of the ProRes is that it supports alpha. So the pixel format has to be YUVA, which means 0-3 would work, but there would be no transparency. ProRes is an intermediate codec not intended for distribution and mainly for editing, so super large file sizes are normal.

If you're interested in a read about alpha channels and codecs, there is a good blog post here.
https://curiosalon.github.io/blog/ffmpeg-alpha-masking/

TLDR: There is no distribution codec besides webm with VP8/VP9 that supports alpha. And VP9 encoding is super slow, and not supported by really any video editors. You either have to have a large file size, or chromakey out a background.

**Comment by Adenser** [comment]
Thanks for the lecture, I guess it's solved. I would think about adding information when out of memory expection pop up it would inform about the small amount of space disk available :D
